assets/src/components/mapPage/routePropertiesCard.tsx

```diff
@@ -0,0 +1,168 @@
+import React from "react"
+import { useRoute } from "../../contexts/routesContext"
+import {
+  RoutePattern,
+  RoutePatternId,
+  DirectionName,
+  DirectionId,
+  RouteId,
+  Route,
+  ByRoutePatternId,
+} from "../../schedule"
+import { Card } from "../card"
+import { RoutePill } from "../routePill"
+
+const sortRoutePatterns = (routePatterns: RoutePattern[]): RoutePattern[] =>
+  routePatterns.sort((rp1, rp2) => rp1.sortOrder - rp2.sortOrder)
+
+const DirectionPicker = ({
+  selectedRoutePattern,
+  routePatterns,
+  selectRoutePattern,
+  directionNames,
+}: {
+  selectedRoutePattern: RoutePattern
+  routePatterns: ByRoutePatternId<RoutePattern>
+  selectRoutePattern: (routePattern: RoutePattern) => void
+  directionNames: {
+    0: DirectionName
+    1: DirectionName
+  }
+}) => {
+  return (
+    <div className="m-route-properties-card__direction-picker">
+      {
+        // TODO: Should we always show both buttons? Some routes only go one direction?
+      }
+      {[0, 1].map((directionId) => (
+        <div key={directionId}>
+          <input
+            type="radio"
+            id={`direction-radio-${directionId}`}
+            name="direction"
+            defaultChecked={selectedRoutePattern.directionId === directionId}
+            onChange={() => {
+              const targetDirectionId = directionId === 0 ? 1 : 0
```

Comment:
**question:** How necessary is this line? I thought it might be to get TypeScript to do type narrowing on `directionId`, but I tried locally removing `targetDirectionId` and using `directionId` directly in the `find` below and it didn't complain.

Reply:
Not entirely necessary - I found it helpful in thinking through the logic to name it, but there is actually a bug here so maybe it wasn't that helpful after all! Since onChange is only called when the radio button becomes checked, this should be targeting the directionId represented by the input, not switching the direction.